case-server/src/main/java/com/powsybl/caseserver/CaseService.java

```diff
@@ -118,6 +130,9 @@ void importCase(MultipartFile mpf) {
             }
             throw e;
         }
+
+        CaseInfos caseInfos = new CaseInfos(caseFile.getFileName().toString(), getFormat(caseFile));
+        caseInfosPublisher.onNext(CaseInfos.getMessage(caseInfos));
```

Comment:
From what I understand, It's a bug to call onNext concurrently like this. Instead we should get the FluxSink (FluxProcessor.sink() or variants with strategy) and use its next methods.

Reply:
did you use https://stackoverflow.com/a/59136155/2748664 as a reference ? They have the bug too I think

Reply:
Hi Jon
I will analyze all your remarks tomorrow

Reply:
Also documented here: https://cloud.spring.io/spring-cloud-static/spring-cloud-stream/3.0.4.RELEASE/reference/html/spring-cloud-stream.html#_using_reactor_api

Reply:
This has an interesting discussion : PR 3100 in https://github.com/spring-projects/spring-integration/pulls